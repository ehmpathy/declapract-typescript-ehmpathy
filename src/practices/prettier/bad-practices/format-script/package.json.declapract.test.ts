import { fix } from './package.json.declapract';
describe('format script package.json', () => {
  it('should replace "format" with "fix:format" in scripts', async () => {
    const exampleFoundContents = `
{
  "scripts": {
    "prior-key": "prior key value",
    "format": "some contents of format",
    "another-key": "another key value"
  },
  "name": "has a name"
}
    `;
    const fixedContents = await fix(exampleFoundContents, {} as any);
    expect(fixedContents.contents).not.toContain('"format"');
    expect(fixedContents.contents).toMatchSnapshot();
  });
});
