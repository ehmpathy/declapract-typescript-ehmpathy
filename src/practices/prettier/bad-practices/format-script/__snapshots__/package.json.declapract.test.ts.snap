// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format script package.json should replace "format" with "fix:format" in scripts 1`] = `
"{
  \\"scripts\\": {
    \\"prior-key\\": \\"prior key value\\",
    \\"another-key\\": \\"another key value\\"
  },
  \\"name\\": \\"has a name\\"
}"
`;
